exts/cesium.omniverse/cesium/omniverse/ui/credits_viewport_controller.py

```diff
@@ -0,0 +1,89 @@
+from .credits_parser import CesiumCreditsParser, ParsedCredit
+from typing import List, Optional, Tuple
+import logging
+import carb.events
+import omni.kit.app as app
+from ..bindings import ICesiumOmniverseInterface
+import omni.ui as ui
+
+
+class CreditsViewportController:
+    _instance = None
+    _cesium_omniverse_interface = None
+    _some_string = None
+    _logger: Optional[logging.Logger] = None
+    _parsed_credits: List[ParsedCredit] = []
+    _credits: List[Tuple[str, bool]] = []
+    _event_handlers = []
+    _subscriptions: List[carb.events.ISubscription] = []
+    _initted: bool = False
```

Comment:
So, Python can be a bit weird. All of these should be instantiated in the `__init__` block, and all `None` values should have a type hint defined.

The way this is done right now will actually declare these as a class variable rather than an instance variable. While on the surface this seems the same, class variables actually share copies between every instance of the class when the class is first initialized, and instance variables are unique values for every class. When you're dealing with arrays and such, this can be a huge footgun if you're not careful. Since this controller is basically acting as a singleton right now we're not running into issues but we should absolutely move these declarations into the `__init__`.

Reply:
I actually moved these out of __init__ so they'd be class variables because I'm thinking like a C#/++ singleton person. But sounds like it's more Pythonic to use a global object, so--yeah--let me move the approach to that.